arch/arm/src/armv7-m/arm_stackcheck.c

```diff
@@ -69,6 +69,12 @@ void __stack_overflow_trap(void)
 
   uint32_t regval;
 
+  regval = getreg32(NVIC_DHCSR);
```

Comment:
It is too late if you stop at the hardfault handler. You do not know the cause of the fault (it may be anything else, not only this trap).

Also stopping before the interrupt call means that the debugger can access the context of the offending task, you can look around and better understand the issue.

Reply:
So set the break here. It uses sooooo much FLASH and CPU We only turn it on to verify a system. The margin is large and somewhat unrealistic.
I set it up to give you the size increases needed  see https://cwiki.apache.org/confluence/display/NUTTX/ARMv7-M+Run+Time+Stack+Checking "As of commit 4942867 Register R11 will..."

Reply:
@davids5 Sorry I did't understand. Were do you suggest to add the breakpoint?

I don't see how this change will cause any overhead. The added code is within the trap, which is only called after a fault has occurred. So, during normal circumstances, it shouldn't affect in any way the function call mechanism.

Reply:
@fjpanag I am suggesting you do not need to add code to add a break point. In the debugger set it on the symbol `__stack_overflow_trap`.

Reply:
@davids5 This is what I am doing till now, but there are two problems:
* There are limited breakpoints that the debugger can set. On more for this check means one less for debugging the application.
* If for any reason you forget / remove it and the check fails, you lost the opportunity to diagnose the issue.

Since this is not going to introduce any overhead, why not add it?

Reply:
Building without this PR:
```
   text	   data	    bss	    dec	    hex	filename
 863483	   3900	  48352	 915735	  df917	bin/firmware
```

Building with this PR:
```
   text	   data	    bss	    dec	    hex	filename
 863499	   3900	  48352	 915751	  df927	bin/firmware
```

This is a 16 byte difference in text only.

Reply:
Maybe a new Kconfig option would satisfy everyone?

Reply:
Well it does add overhead and Unlimited Flash Breakpoint are available on Segger's Jlink. 

The Build is soo big that that 16 bytes could break it. 

Kconfig works for me.

Reply:
@davids5 Added Kconfig to enable this feature. By default it is disabled, so the build remains as it was before.